File: tests_deps/src/main/java/com/github/klee0kai/test/car/model/CarInject.java
package com.github.klee0kai.test.car.model;

import javax.inject.Inject;

public class CarInject {


    @Inject
    public Bumper bumper;

    @Inject
    public Wheel wheel;

    @Inject
    public Window window;

    public Bumper bumperFromMethod;

    public Wheel wheelFromMethod;

    public Window windowFromMethod;


    @Inject
    public void init(Bumper bumper, Wheel wheel, Window window) {
        bumperFromMethod = bumper;
        wheelFromMethod = wheel;
        windowFromMethod = window;
    }

}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/model/CarInject.kt
package com.github.klee0kai.test.car.model

import javax.inject.Inject

class CarInject {
    @Inject
    var bumper: Bumper? = null

    @Inject
    var wheel: Wheel? = null

    @Inject
    var window: Window? = null


    var bumperFromMethod: Bumper? = null
    var wheelFromMethod: Wheel? = null
    var windowFromMethod: Window? = null

    @Inject
    fun init(bumper: Bumper?, wheel: Wheel?, window: Window?) {
        bumperFromMethod = bumper
        wheelFromMethod = wheel
        windowFromMethod = window
    }
}
